spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -324,6 +330,41 @@ def main(argv=None):
         dim = dim_list.index(dim)
         im_out = split_data(im_in, dim)
 
+    elif arguments.stitch is not None:
+        # preserve original orientation
+        orig_ornt = im_in_list[0].orientation
+        # reorient images to PIL to use stitching module
+        im_ornt = []
+
+        # use SCT to reorient images
+        for idx, i in enumerate(arguments.i):
+            # TODO: replace 0.nii.gz, etc. by temp files
+            # TODO: where does SCT save this, temp dir?
+            cmd = ['sct_image', '-i', str(i), f"-o {idx}.nii.gz", '-setorient', 'RPI']
+            im_ornt.append(f"{idx}.nii.gz")
+            status, output = run_proc(cmd, verbose=verbose, is_sct_binary=True)
```

Comment:
Hey @joshuacwnewton,

you're way too kind, thank you so much for all the hints and details, this makes it lots of fun to contribute to SCT! :1st_place_medal:
I have been really enjoying it! :slightly_smiling_face: 

With respect to the proc calls, thank you very much for the helpful input, I tried to address this in [f7d037d](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3865/commits/f7d037dfb0f211fdc134abe4af6d467173e30432)

Reply:
Ah, my apologies! I was a little unclear about what I meant by an "internal function". In this case, `sct_image.main()` isn't exactly an "internal function", it's still basically calling the SCT script.

To replace `sct_image.main()` with the internal function, we can look at what `-setorient` does here:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/bd3d89902106be3186fe051096c72814a95982ce/spinalcordtoolbox/scripts/sct_image.py#L305-L307

If we ignore the `printv` call, we can see that `-setorient` is just a simple function call, so we can just import the `change_orientation()` function itself and use it directly:

```python
from spinalcordtoolbox.image import Image, change_orientation

[...]

im_in = Image(str(i))
im_out = change_orientation(im_in, 'RPI')
```

In this case, instead of working with filename strings, we would be working with `Image` objects.

Reply:
> If we ignore the `printv` call, we can see that `-setorient` is just a simple function call, so we can just import the `change_orientation()` function itself and use it directly:
> 
> ```python
> from spinalcordtoolbox.image import Image, change_orientation
> 
> [...]
> 
> im_in = Image(str(i))
> im_out = change_orientation(im_in, 'RPI')
> ```
> 
> In this case, instead of working with filename strings, we would be working with `Image` objects.

Perfect example, thank you so much, I think this should be implemented in a similar way now :+1: c.f. 9d88012